packages/@glimmer/compiler/lib/compiler.ts

```diff
@@ -53,25 +54,200 @@ const defaultOptions: PrecompileOptions = {
  *     let template = factory.create(env);
  *
  * @method precompile
- * @param {string} string a Glimmer template string
+ * @param {string} source a Glimmer template string
  * @return {string} a template javascript string
  */
-export function precompile(string: string, options?: PrecompileOptions): TemplateJavascript;
+export function precompile(source: string, options?: PrecompileOptions): TemplateJavascript;
 export function precompile(
-  string: string,
+  source: string,
   options: PrecompileOptions = defaultOptions
 ): TemplateJavascript {
-  let ast = preprocess(string, options);
+  let ast = preprocess(source, options);
   let { meta } = options;
-  let { block } = TemplateCompiler.compile(ast, string, options);
+  let { block } = TemplateCompiler.compile(ast, source, options);
+
+  let blockJSON = block.toJSON();
+  let blockJSONString = JSON.stringify(blockJSON);
+
   let idFn = options.id || defaultId;
-  let blockJSON = JSON.stringify(block.toJSON());
-  let templateJSONObject: SerializedTemplateWithLazyBlock<unknown> = {
-    id: idFn(JSON.stringify(meta) + blockJSON),
-    block: blockJSON,
-    meta,
-  };
-
-  // JSON is javascript
-  return JSON.stringify(templateJSONObject);
+  let id = idFn(JSON.stringify(meta) + blockJSONString);
+
+  let debug = new WireFormatDebugger(blockJSON);
+
+  let obj = new ObjectCompiler();
+
+  obj.set('id', id);
+  obj.set('meta', meta);
+
+  let func = new FunctionCompiler(obj.indentation);
+
+  func.push('/*');
+  func.push();
+
+  func.push('Source:');
+  func.push();
+  func.push('```hbs');
+  func.push(source);
+  func.push('```');
+  func.push();
+
+  func.push('Transformed:');
+  func.push();
+  func.push('```hbs');
+  func.push(print(ast));
+  func.push('```');
+  func.push();
+
+  let wf = new ObjectCompiler(func.indentation);
+
+  wf.set('symbols', blockJSON.symbols);
+  wf.set('hasEval', blockJSON.hasEval);
+  wf.set('upvars', blockJSON.upvars);
+
+  let stmts = new WireFormatArrayCompiler(wf.indentation);
+
+  blockJSON.statements.forEach(stmt => {
+    stmts.add(stmt, debug.formatOpcode(stmt));
+  });
+
+  wf.setRaw('statements', stmts.compile());
+
+  if (!DEBUG) {
```

Comment:
Instead of using the imported DEBUG, can you add an option? This would allow us to compile both production and development templates from the same actual build (and is how I intend to "fix" the AST transforms in Ember to avoid debug only AST transforms from hitting prod).